sync_event is experimental like SyncEvent and SyncManager

sidewayss opened the pull request:
The docs say the ServiceWorkerGlobalScope ["sync" event](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerGlobalScope/sync_event) is type [SyncEvent](https://developer.mozilla.org/en-US/docs/Web/API/SyncEvent) and depends on [SyncManager](https://developer.mozilla.org/en-US/docs/Web/API/SyncManager), and both of those are experimental technology, so the "sync" event must be experimental too.

NOTE: the script has been flipped and now `SyncEvent` and `SyncManager` are *not* experimental.  This PR reflects that change.

skyclouds2001 commented:
per https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#setting-experimental, the Background Sync API could be mark as non-experiment, as they are supported by at least one platform and for more then two years (in fact it is eight years).

skyclouds2001 commented:
Yes, you can change this PR to label both `SyncEvent` and `SyncManager` (and `ServiceWorkerRegistration.sync`) are not experimental

queengooborg commented:
Yes, per the BCD guidelines, these features should be marked as non-experimental.  @sidewayss, would you be down to change this PR (or create a new one) to mark `SyncEvent` and `SyncManager` as non-experimental instead?

sidewayss commented:
> @sidewayss, would you be down to change this PR (or create a new one) to mark `SyncEvent` and `SyncManager` as non-experimental instead?

Sure, I'll try to get to it later today.

sidewayss commented:
I modified this branch to remove my original change and make `SyncEvent` and `SyncManager` *not* experimental. I removed the experimental property from the SyncEvent constructor, but I'm not sure which other properties/methods of both of these to mark as non-experimental.  That will require some review from you, @queengooborg.  Feel free to make whatever changes those are and merge, otherwise tell me and I'll make them.  Thanks.

skyclouds2001 commented:
[`ServiceWorkerRegistration.sync`](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/sync) can also mark as non-experimental, too

sidewayss commented:
I have tagged [ServiceWorkerRegistration.sync](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/sync) as not experimental.
